src/lib/resolve-ipld-path.js

```diff
@@ -64,37 +67,45 @@ export default async function resolveIpldPath (ipld, sourceCid, path, nodes = []
   if (link) {
     pathBoundaries.push(link)
     // Go again, using the link.target as the sourceCid, and the remainderPath as the path.
-    return resolveIpldPath(ipld, link.target, remainderPath, nodes, pathBoundaries)
+    return resolveIpldPath(ipfs, link.target, remainderPath, nodes, pathBoundaries)
   }
   // we made it to the containing node. Hand back the info
   const canonicalPath = path ? `${sourceCidStr}${path}` : sourceCidStr
   const targetNode = node
+
   return { targetNode, canonicalPath, localPath: path, nodes, pathBoundaries }
 }
 
 /**
  * @function ipldGetNodeAndRemainder
- * @param {IpldInterface} ipld
+ * @param {typeof import('@helia/interface').Helia} helia
  * @param {string} sourceCid
  * @param {string} path
  * @returns
  */
-export async function ipldGetNodeAndRemainder (ipld, sourceCid, path) {
+export async function ipldGetNodeAndRemainder (helia, sourceCid, path) {
   if (sourceCid == null) {
     throw new Error('sourceCid is null')
   }
-  // TODO: find out why ipfs.dag.get with localResolve never resolves.
-  // const {value, remainderPath} = await getIpfs().dag.get(sourceCid, path, {localResolve: true})
+  let cidInstance = CID.asCID(sourceCid)
+  if (cidInstance === null) {
+    cidInstance = CID.parse(sourceCid)
+  }
+  const codecWrapper = await getCodecForCid(cidInstance)
+  const encodedValue = await getRawBlock(helia, cidInstance)
```

Comment:
instead of passing ipfs/helia through to this, we should pass an object:

```js
{
  getBlock: (cidInstance) => Promise<Uint8Array>
}
```

I feel there were other methods we need to support, but if not, we can simply pass the function itself.

That fn would be a wrapper for kubo-rpc-client.dag.get in webui/desktop, and helia.blockstore.get in explore.ipld.io